janus/__init__.py

```diff
@@ -41,6 +41,19 @@ class ShutDown(Exception):
     "BaseQueue",
 )
 
+_contra_T = TypeVar("_contra_T", contravariant=True)
+
+class SupportsLT(Protocol[_contra_T]):
+    def __lt__(self, other: _contra_T):  ...
```

Comment:
Since we're only using these with Any you could simplify these to non-generic protocols that just have `other: Any`.

Reply:
There's an argument not to, but I don't think any existing typechecker implements a rule checking for variance compatability, nor is it specified, I'll go ahead with the simplification.